lib/tools/src/cover.erl

```diff
@@ -2704,8 +2707,20 @@ analyse_list_to_file(Modules, Opts, State) ->
 		  OutFile = outfilename(OutDir,Module,HTML),
 		  do_analyse_to_file(Module,File,OutFile,HTML,State)
 	  end,
-    {Ok,Error1} = split_ok_error(pmap(Fun, LoadedMF++ImportedMF),[],[]),
-    {result,Ok,Error ++ Error1}.
+
+    MFs = LoadedMF ++ ImportedMF,
+    {AnalyseOk, AnalyseError} = split_ok_error(pmap(Fun, MFs), [], []),
+
+    %% Don't generate a summary file if only one module is analysed
+    {Ok, Error1} =
+        if length(MFs) > 1 ->
+               SummaryRes = do_summary(MFs, Opts, State),
+               split_ok_error(SummaryRes, AnalyseOk, AnalyseError);
+           true ->
+               {AnalyseOk, AnalyseError}
+        end,
```

Comment:
Regarding not emitting the summary report if explicitly analysing only one
file, perhaps it should be enabled by default instead of omitted.

The information in the summary is not readily available in any other way
without manually analysing with e.g. cover:analyse/0 to get covered,
not covered lines, and then calculating the coverage percentage.